jobs/ScheduleBackupJob.js

```diff
@@ -80,11 +80,42 @@ class ScheduleBackupJob extends BaseJob {
   }
 
   static deleteOldBackup(job, instanceDeleted) {
+    let transactionLogsDeleteBefore;
+
+    function filterOldBackups(oldBackups) {
+      let filteredOldBackups = [];
+      /* oldBackups : This aray should contain all older backups
+      // including last retenion day's backup. E.g. if retention period
+      // is 15 days it would include all backups in (15th, 16th, 17th ...) */
+      if (typeof oldBackups !== 'undefined' && oldBackups.length > 0) {
+        // Older backups are sorted as latest at first
+        let sortedBackups = _.sortBy(oldBackups, ['started_at']).reverse();
+        const latestSuccessIndex = _.findIndex(sortedBackups,
+          backup => backup.state === CONST.OPERATION.SUCCEEDED);
+        if (latestSuccessIndex === -1) {
+          //No successful backup beyond retention period.
+          filteredOldBackups = sortedBackups;
+          transactionLogsDeleteBefore = new Date(Date.now() - (config.backup.retention_period_in_days + 1) * 24 * 60 * 60 * 1000).toISOString();
+        } else {
+          //Should return backups before a successful backup.
+          filteredOldBackups = _.slice(sortedBackups, latestSuccessIndex + 1);
+          transactionLogsDeleteBefore = _.get(sortedBackups[latestSuccessIndex], 'started_at');
+        }
+      }
+      return filteredOldBackups;
+    }
+
     const options = _.omit(job.attrs.data, 'trigger', 'type');
+    const serviceName = catalog.getService(job.attrs.data.service_id).name;
+    const listOptions = {
+      prefix: `${serviceName}/logs/${job.attrs.data.instance_id}`
+    };
+    const container = `${backupStore.containerPrefix}-${serviceName}`;
```

Comment:
As commented above, both prefix and container formation can be / should be part of BackupStore.